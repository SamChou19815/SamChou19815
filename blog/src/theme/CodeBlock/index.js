/**
 * Copyright (c) 2019-present, Developer Sam.
 *
 * This source code is licensed under the AGPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import CodeBlock from 'sam-highlighter/CodeBlock';
import initialize from 'sam-highlighter/language';

import styles from './styles.module.css';

initialize();

export default ({ children, className: languageClassName }) => {
  const language = languageClassName && languageClassName.replace(/language-/, '');
  return (
    <CodeBlock language={language} className={styles.CodeBlock}>
      {children}
    </CodeBlock>
  );
};
