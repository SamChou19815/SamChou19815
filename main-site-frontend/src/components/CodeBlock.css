/* The syntax highlighting for the code block. */

.hljs {
  display: block;
  overflow-x: auto;
  padding: .5em;
  color: #383a42;
  background: #fafafa;
}

.hljs-comment, .hljs-quote {
  color: #808080;
  font-style: italic
}

.hljs-doctag, .hljs-keyword, .hljs-formula, .hljs-built_in {
  color: #3E7AE2;
  font-weight: bold;
}

.hljs-class, .hljs-type {
  color: #9A30AD;
}

.hljs-type-params {
  color: #20999D;
}

.hljs-section, .hljs-name, .hljs-selector-tag, .hljs-deletion, .hljs-subst {
  color: #e45649
}

.hljs-literal {
  color: #0184bb
}

.hljs-string, .hljs-regexp, .hljs-addition, .hljs-attribute, .hljs-meta-string {
  color: #1A8F52;
}

.hljs-attr,
.hljs-variable,
.hljs-template-variable,
.hljs-selector-class,
.hljs-selector-attr,
.hljs-selector-pseudo,
.hljs-number {
  color: #986801
}

.hljs-symbol, .hljs-bullet, .hljs-link, .hljs-meta, .hljs-selector-id, .hljs-title {
  color: #4078f2
}

.hljs-emphasis {
  font-style: italic
}

.hljs-strong {
  font-weight: bold
}

.hljs-link {
  text-decoration: underline
}
