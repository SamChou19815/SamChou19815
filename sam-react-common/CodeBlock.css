/* The syntax highlighting for the code block. */

.hljs {
  display: block;
  overflow-x: auto;
  padding: 0.5em;
  color: #383a42;
  background: #f7f7f7;
}

.hljs-comment,
.hljs-quote {
  color: #808080;
  font-style: italic;
}

.hljs-doctag,
.hljs-keyword,
.hljs-formula,
.hljs-built_in {
  color: #3e7ae2;
  font-weight: 500;
}

.hljs-class,
.hljs-type,
.hljs-title {
  color: #9a30ad;
}

.hljs-method {
  color: #d52262;
}

.hljs-section,
.hljs-name,
.hljs-selector-tag,
.hljs-deletion,
.hljs-subst {
  color: #e45649;
}

.hljs-literal {
  color: #0184bb;
}

.hljs-string,
.hljs-regexp,
.hljs-addition,
.hljs-attribute,
.hljs-meta-string {
  color: #1a8f52;
}

.hljs-attr {
  color: #3e7ae2;
}

.hljs-variable,
.hljs-template-variable,
.hljs-selector-class,
.hljs-selector-attr,
.hljs-selector-pseudo,
.hljs-number {
  color: #986801;
}

.hljs-symbol,
.hljs-bullet,
.hljs-link,
.hljs-meta,
.hljs-selector-id {
  color: #4078f2;
}

.hljs-emphasis {
  font-style: italic;
}

.hljs-strong {
  font-weight: bold;
}

.hljs-link {
  text-decoration: underline;
}
