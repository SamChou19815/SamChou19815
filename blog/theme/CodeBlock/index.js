/**
 * Copyright (c) 2017-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import CodeBlock, { initialize } from 'sam-highlighter';

import styles from './styles.module.css';

initialize();

export default ({ children, className: languageClassName }) => {
  const language = languageClassName && languageClassName.replace(/language-/, '');
  return (
    <CodeBlock language={language} className={styles.CodeBlock}>
      {children}
    </CodeBlock>
  );
};
