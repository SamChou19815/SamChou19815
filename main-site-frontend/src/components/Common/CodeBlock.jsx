// @flow strict

import type { Node } from 'react';
import React from 'react';
// $FlowFixMe
import highlight from 'highlight.js';
import './CodeBlock.css';

type Props = {|
  +children: string;
  +className?: string;
  +style?: { +[string]: string }
|};

export default function CodeBlock({ children, className, style }: Props): Node {
  const codeBlockRef = React.useRef(null);
  React.useEffect(() => highlight.highlightBlock(codeBlockRef.current));
  return (
    <pre className={className} style={style}>
      <code className="samlang" ref={codeBlockRef}>
        {children}
      </code>
    </pre>
  );
}
