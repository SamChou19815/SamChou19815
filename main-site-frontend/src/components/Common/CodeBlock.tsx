import React, { ReactElement } from 'react';
import highlight from 'highlight.js';
import './CodeBlock.css';

type Props = {
  readonly children: string;
  readonly className?: string;
  readonly style?: { [cssProp: string]: string }
};

export default function CodeBlock({ children, className, style }: Props): ReactElement {
  const codeBlockRef = React.useRef(null);
  React.useEffect(() => {
    const node = codeBlockRef.current;
    if (node == null) {
      return;
    }
    try {
      highlight.highlightBlock(node);
    } catch (e) {
      // eslint-disable-next-line no-console
      console.log('Safari does not support lookbehind regex. Skip rerender!');
    }
  });
  return (
    <pre className={className} style={style}>
      <code className="samlang" ref={codeBlockRef}>
        {children}
      </code>
    </pre>
  );
}
