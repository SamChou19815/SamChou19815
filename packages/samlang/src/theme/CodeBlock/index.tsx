/**
 * Copyright (c) 2019-present, Developer Sam.
 *
 * This source code is licensed under the AGPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';

import CodeBlock from 'lib-react/CodeBlock';
import initialize from 'lib-react/language';

import styles from './styles.module.css';

initialize();

type Props = { readonly children: string; readonly className?: string };

export default ({ children, className: languageClassName }: Props) => {
  const language = languageClassName && languageClassName.replace(/language-/, '');
  return (
    <CodeBlock language={language} className={styles.CodeBlock}>
      {children}
    </CodeBlock>
  );
};
