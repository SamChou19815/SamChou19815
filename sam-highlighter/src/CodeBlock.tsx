import React, { CSSProperties, ReactElement } from 'react';
import highlight from 'highlight.js';
import './CodeBlock.css';

type Props = {
  readonly language: string;
  readonly children: string;
  readonly className?: string;
  readonly style?: CSSProperties;
};

export default ({ language, children, className, style }: Props): ReactElement => {
  const codeBlockRef = React.useRef(null);
  React.useEffect((): void => {
    const node = codeBlockRef.current;
    if (node == null) {
      return;
    }
    try {
      highlight.highlightBlock(node);
    } catch (e) {
      // eslint-disable-next-line no-console
      console.log('Safari does not support lookbehind regex. Skip rerender!');
    }
  });
  return (
    <pre className={className} style={style}>
      <code className={language} ref={codeBlockRef}>
        {children}
      </code>
    </pre>
  );
}
