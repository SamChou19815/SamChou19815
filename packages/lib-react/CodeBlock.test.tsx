import React from 'react';

import renderer from 'react-test-renderer';

import CodeBlock from 'lib-react/PrismCodeBlock';

const createSamLangTest = (name: string, code: string) => {
  it(`CodeBlock rendering "${name}" matches snapshot.`, () => {
    const tree = renderer.create(<CodeBlock language="samlang">{code}</CodeBlock>).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

// Examples from samlang.developersam.com

const helloWorld = `
class Main {
  function main(): string = "Hello World"
}
`;

const fortyTwo = `
class Main {
  function main(): int = 2 * 21
}
`;

const patternMatching = `
class Option<T>(None(unit), Some(T)) {

  public function <T> getNone(): Option<T> = None(unit)

  public function <T> getSome(d: T): Option<T> = Some(d)

  public method <R> map(f: (T) -> R): Option<R> =
      match (this) {
          | None _ -> None(unit)
          | Some d -> Some(f(d))
      }

}
`;

const typeInference = `
class TypeInference {

  function notAnnotated(): unit =
      val _ = (a, b, c) -> if a(b + 1) then b else c;

  // Read the docs to see how we do the type inference.
  function annotated(): unit =
      val _: ((int) -> bool, int, int) -> int =
          (a: (int) -> bool, b: int, c: int) -> (
              if a(b + 1) then b else c
          );

}
`;

createSamLangTest('Hello World', helloWorld);
createSamLangTest('42', fortyTwo);
createSamLangTest('Pattern Matching', patternMatching);
createSamLangTest('Type Inference', typeInference);
