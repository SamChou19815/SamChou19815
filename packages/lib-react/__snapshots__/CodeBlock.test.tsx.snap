// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeBlock rendering "42" matches snapshot. 1`] = `
<pre
  className="prism-code language-samlang"
  style={
    Object {
      "backgroundColor": "#f7f7f7",
    }
  }
>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      class
    </span>
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      Main
    </span>
    <span
      className="token plain"
    >
       {
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
        
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      function
    </span>
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token function"
      style={
        Object {
          "color": "rgb(213, 34, 98)",
        }
      }
    >
      main
    </span>
    <span
      className="token plain"
    >
      (): 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      int
    </span>
    <span
      className="token plain"
    >
       = 2 
    </span>
    <span
      className="token operator"
      style={
        Object {
          "color": "rgb(56, 72, 79)",
        }
      }
    >
      *
    </span>
    <span
      className="token plain"
    >
       21
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
      }
    </span>
  </div>
</pre>
`;

exports[`CodeBlock rendering "Hello World" matches snapshot. 1`] = `
<pre
  className="prism-code language-samlang"
  style={
    Object {
      "backgroundColor": "#f7f7f7",
    }
  }
>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      class
    </span>
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      Main
    </span>
    <span
      className="token plain"
    >
       {
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
        
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      function
    </span>
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token function"
      style={
        Object {
          "color": "rgb(213, 34, 98)",
        }
      }
    >
      main
    </span>
    <span
      className="token plain"
    >
      (): 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      string
    </span>
    <span
      className="token plain"
    >
       = 
    </span>
    <span
      className="token string"
      style={
        Object {
          "color": "rgb(26, 143, 82)",
        }
      }
    >
      "Hello World"
    </span>
    <span
      className="token plain"
    >
      
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
      }
    </span>
  </div>
</pre>
`;

exports[`CodeBlock rendering "Pattern Matching" matches snapshot. 1`] = `
<pre
  className="prism-code language-samlang"
  style={
    Object {
      "backgroundColor": "#f7f7f7",
    }
  }
>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      class
    </span>
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      Option
    </span>
    <span
      className="token plain"
    >
      &lt;
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      T
    </span>
    <span
      className="token plain"
    >
      &gt;(
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      None
    </span>
    <span
      className="token plain"
    >
      (
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      unit
    </span>
    <span
      className="token plain"
    >
      ), 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      Some
    </span>
    <span
      className="token plain"
    >
      (
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      T
    </span>
    <span
      className="token plain"
    >
      )) {
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
        public 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      function
    </span>
    <span
      className="token plain"
    >
       &lt;
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      T
    </span>
    <span
      className="token plain"
    >
      &gt; 
    </span>
    <span
      className="token function"
      style={
        Object {
          "color": "rgb(213, 34, 98)",
        }
      }
    >
      getNone
    </span>
    <span
      className="token plain"
    >
      (): 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      Option
    </span>
    <span
      className="token plain"
    >
      &lt;
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      T
    </span>
    <span
      className="token plain"
    >
      &gt; = 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      None
    </span>
    <span
      className="token plain"
    >
      (
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      unit
    </span>
    <span
      className="token plain"
    >
      )
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
        public 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      function
    </span>
    <span
      className="token plain"
    >
       &lt;
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      T
    </span>
    <span
      className="token plain"
    >
      &gt; 
    </span>
    <span
      className="token function"
      style={
        Object {
          "color": "rgb(213, 34, 98)",
        }
      }
    >
      getSome
    </span>
    <span
      className="token plain"
    >
      (d: 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      T
    </span>
    <span
      className="token plain"
    >
      ): 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      Option
    </span>
    <span
      className="token plain"
    >
      &lt;
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      T
    </span>
    <span
      className="token plain"
    >
      &gt; = 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      Some
    </span>
    <span
      className="token plain"
    >
      (d)
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
        public 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      method
    </span>
    <span
      className="token plain"
    >
       &lt;
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      R
    </span>
    <span
      className="token plain"
    >
      &gt; 
    </span>
    <span
      className="token function"
      style={
        Object {
          "color": "rgb(213, 34, 98)",
        }
      }
    >
      map
    </span>
    <span
      className="token plain"
    >
      (f: (
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      T
    </span>
    <span
      className="token plain"
    >
      ) 
    </span>
    <span
      className="token operator"
      style={
        Object {
          "color": "rgb(56, 72, 79)",
        }
      }
    >
      -
    </span>
    <span
      className="token plain"
    >
      &gt; 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      R
    </span>
    <span
      className="token plain"
    >
      ): 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      Option
    </span>
    <span
      className="token plain"
    >
      &lt;
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      R
    </span>
    <span
      className="token plain"
    >
      &gt; =
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
            
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      match
    </span>
    <span
      className="token plain"
    >
       (
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      this
    </span>
    <span
      className="token plain"
    >
      ) {
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
                | 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      None
    </span>
    <span
      className="token plain"
    >
       _ 
    </span>
    <span
      className="token operator"
      style={
        Object {
          "color": "rgb(56, 72, 79)",
        }
      }
    >
      -
    </span>
    <span
      className="token plain"
    >
      &gt; 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      None
    </span>
    <span
      className="token plain"
    >
      (
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      unit
    </span>
    <span
      className="token plain"
    >
      )
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
                | 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      Some
    </span>
    <span
      className="token plain"
    >
       d 
    </span>
    <span
      className="token operator"
      style={
        Object {
          "color": "rgb(56, 72, 79)",
        }
      }
    >
      -
    </span>
    <span
      className="token plain"
    >
      &gt; 
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      Some
    </span>
    <span
      className="token plain"
    >
      (
    </span>
    <span
      className="token function"
      style={
        Object {
          "color": "rgb(213, 34, 98)",
        }
      }
    >
      f
    </span>
    <span
      className="token plain"
    >
      (d))
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
            }
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
      }
    </span>
  </div>
</pre>
`;

exports[`CodeBlock rendering "Type Inference" matches snapshot. 1`] = `
<pre
  className="prism-code language-samlang"
  style={
    Object {
      "backgroundColor": "#f7f7f7",
    }
  }
>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      class
    </span>
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token class-name"
      style={
        Object {
          "color": "rgb(154, 48, 173)",
        }
      }
    >
      TypeInference
    </span>
    <span
      className="token plain"
    >
       {
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
        
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      function
    </span>
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token function"
      style={
        Object {
          "color": "rgb(213, 34, 98)",
        }
      }
    >
      notAnnotated
    </span>
    <span
      className="token plain"
    >
      (): 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      unit
    </span>
    <span
      className="token plain"
    >
       =
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
            
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      val
    </span>
    <span
      className="token plain"
    >
       _ = (a, b, c) 
    </span>
    <span
      className="token operator"
      style={
        Object {
          "color": "rgb(56, 72, 79)",
        }
      }
    >
      -
    </span>
    <span
      className="token plain"
    >
      &gt; 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      if
    </span>
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token function"
      style={
        Object {
          "color": "rgb(213, 34, 98)",
        }
      }
    >
      a
    </span>
    <span
      className="token plain"
    >
      (b 
    </span>
    <span
      className="token operator"
      style={
        Object {
          "color": "rgb(56, 72, 79)",
        }
      }
    >
      +
    </span>
    <span
      className="token plain"
    >
       1) 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      then
    </span>
    <span
      className="token plain"
    >
       b 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      else
    </span>
    <span
      className="token plain"
    >
       c;
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
        
    </span>
    <span
      className="token comment"
      style={
        Object {
          "color": "rgb(111, 111, 111)",
          "fontStyle": "italic",
        }
      }
    >
      // Read the docs to see how we do the type inference.
    </span>
    <span
      className="token plain"
    >
      
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
        
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      function
    </span>
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token function"
      style={
        Object {
          "color": "rgb(213, 34, 98)",
        }
      }
    >
      annotated
    </span>
    <span
      className="token plain"
    >
      (): 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      unit
    </span>
    <span
      className="token plain"
    >
       =
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
            
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      val
    </span>
    <span
      className="token plain"
    >
       _: ((
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      int
    </span>
    <span
      className="token plain"
    >
      ) 
    </span>
    <span
      className="token operator"
      style={
        Object {
          "color": "rgb(56, 72, 79)",
        }
      }
    >
      -
    </span>
    <span
      className="token plain"
    >
      &gt; 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      bool
    </span>
    <span
      className="token plain"
    >
      , 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      int
    </span>
    <span
      className="token plain"
    >
      , 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      int
    </span>
    <span
      className="token plain"
    >
      ) 
    </span>
    <span
      className="token operator"
      style={
        Object {
          "color": "rgb(56, 72, 79)",
        }
      }
    >
      -
    </span>
    <span
      className="token plain"
    >
      &gt; 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      int
    </span>
    <span
      className="token plain"
    >
       =
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
                (a: (
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      int
    </span>
    <span
      className="token plain"
    >
      ) 
    </span>
    <span
      className="token operator"
      style={
        Object {
          "color": "rgb(56, 72, 79)",
        }
      }
    >
      -
    </span>
    <span
      className="token plain"
    >
      &gt; 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      bool
    </span>
    <span
      className="token plain"
    >
      , b: 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      int
    </span>
    <span
      className="token plain"
    >
      , c: 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      int
    </span>
    <span
      className="token plain"
    >
      ) 
    </span>
    <span
      className="token operator"
      style={
        Object {
          "color": "rgb(56, 72, 79)",
        }
      }
    >
      -
    </span>
    <span
      className="token plain"
    >
      &gt; (
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
                    
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      if
    </span>
    <span
      className="token plain"
    >
       
    </span>
    <span
      className="token function"
      style={
        Object {
          "color": "rgb(213, 34, 98)",
        }
      }
    >
      a
    </span>
    <span
      className="token plain"
    >
      (b 
    </span>
    <span
      className="token operator"
      style={
        Object {
          "color": "rgb(56, 72, 79)",
        }
      }
    >
      +
    </span>
    <span
      className="token plain"
    >
       1) 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      then
    </span>
    <span
      className="token plain"
    >
       b 
    </span>
    <span
      className="token keyword"
      style={
        Object {
          "color": "rgb(62, 122, 226)",
        }
      }
    >
      else
    </span>
    <span
      className="token plain"
    >
       c
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
                );
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      
    </span>
  </div>
  <div
    className="token-line"
    style={
      Object {
        "backgroundColor": null,
      }
    }
  >
    <span
      className="token plain"
    >
      }
    </span>
  </div>
</pre>
`;
